src/BudgetBadgerWebApi.Api/Filters/Common/ResponseWrapObject.cs

```diff
@@ -0,0 +1,9 @@
+﻿namespace BudgetBadgerWebApi.Api.Filters.Common
+{
+    public class ResponseWrapObject
+    {
+        public bool Success { get; set; }
+        public object Value { get; set; } = new();
```

Comment:
It's done this way so we won't have "object not set to an instance of an object" when working with object of the ResponseWrapObject class. Also, middleware will set Success with true/false value and put content to one of the objects, depends of the value of Success property. It will be useful when building client for WebAPI.

Reply:
But why we do not want this type of error? IMO better to have  "object not set to an instance of an object" if we forget something or middleware will be fucked up, instead of passing with umpty object and then wondering why it's not working / why I have empty object here

Reply:
Because this makes sense for one of this objects to be null.

An example of what we can get from WebAPI:
```json
{
  "success": true,
  "value": {
    "id": 1,
    "name": "warzywa i owoce"
  },
  "error": null
}
```

or

```json
{
  "success": false,
  "value": null,
  "error": {
    "title": "Entity not found",
    "details": [
      "Entity with the given id could not be found."
    ]
  }
}
```

Client (for example Blazor app) can have one model for response and just check is `success` is `true` or `false`. Next it can take data from `value` or `error`, based on the value of `success`. This way getting a response and parse it is very logical and standardized across all endpoints.

So it's not for internal use in WebAPI but for cleints that will be sending requests to our app.